plugins/node/opentelemetry-instrumentation-fastify/src/instrumentation.ts

```diff
@@ -96,8 +96,9 @@ export class FastifyInstrumentation extends InstrumentationBase {
       const anyRequest = request as any;
 
       const rpcMetadata = getRPCMetadata(context.active());
-      const routeName =
-        anyRequest.routeOptions?.config?.url || request.routerPath;
+      const routeName = anyRequest.routeOptions
+        ? anyRequest.routeOptions.url
+        : request.routerPath;
```

Comment:
Regarding the codecov warning on this line and below: TAV tests *could* cover line 101, but currently they won't because it doesn't go back to an early-enough fastify version. Shall I add an `// istanbul ignore next` directive here?

Reply:
Interesting; AFAIK codecov uses the reports generated by istanbul, so adding the directive *should* help. Looks like the uncovered lines above are actually not tested at the moment ([see full file on the codecov report](https://app.codecov.io/gh/open-telemetry/opentelemetry-js-contrib/pull/1763/blob/plugins/node/opentelemetry-instrumentation-fastify/src/instrumentation.ts) - not sure if you have access). We could also add an earlier fastify version to TAV if that fixes the issue. 

I've just recently added TAV to this package, having more versions tested would likely be a good idea anyway.

Reply:
I've added testing of fastify@4.0.0 plus the current latest and any subsequent fastify 4.x releases to the .tav.yml.
I don't know if coverage data from TAV runs get included in the codecov summary.

Reply:
Ah right, it looks like reports from TAV are not included. 

IMO just having the version tested is good enough for now, we may look into adding coverage from TAV in the future, but we've been having problems with codecov for a while now, so I think this will likely be a larger task.